dyn_em/module_initialize_real.F

```diff
@@ -181,6 +184,8 @@ SUBROUTINE init_domain_rk ( grid &
                                 ipsy, ipey, jpsy, jpey, kpsy, kpey )
       its = ips ; ite = ipe ; jts = jps ; jte = jpe ; kts = kps ; kte = kpe
 
+      
+      qnwfa_scale_fac = config_flags%nssl_aero_scale
```

Comment:
@dudhia @weiwangncar @kkeene44 @MicroTed 
Folks,
Again, the scenario that someone runs the real program with Thompson and then later changes the namelist.input file to NSSL scheme when running the model, which is discussed in a previous comment. Maybe we add in a input_wrf.F and just stop if there is a problem?

We would need to have the qn[iw]fa_scale_fac be metadata from real to WRF. Then we compare the MP scheme (namelist) and the qn[iw]fa_scale_fac (metadata) to see if they are consistent. This is similar to what we do with grid distance, for example: you can't change it between real and WRF.

Reply:
Oh, but I think I didn't quite understand your point at first, i.e., if somebody runs real.exe with Thompson aero and then changes mp_physics to NSSL but doesn't rerun real.exe. In my current code, that would result in qnn being uninitialized, so that field could be tested for being all zero. If that test fails then one could abort?

Reply:
@MicroTed 
A cleaner way that looking at a bunch of zeros is for us to have some metadata that goes between the real program and the WRF model, in just the same way that we ensure that the grid distance has not changed. Here's an idea for input_wrf.F (this is inside of an ARW only ifdef):
```
 453        !  Test to make sure that the MP aerosol option is consistent.
 454 
 455        IF ( switch .EQ. input_only ) THEN
 456           CALL wrf_get_dom_ti_integer ( fid , 'AERO_OPT' ,  aero_opt_compare , 1 , icnt , ierr )
 457           IF ( aero_opt_compare .NE. config_flags%aero_opt ) THEN
 458              WRITE(wrf_err_message,*)'aero_opt from input file  ',aero_opt_compare
 459              CALL wrf_message(wrf_err_message)
 460              WRITE(wrf_err_message,*)'aero_opt from namelist    ',config_flags%aero_opt
 461              CALL wrf_message(wrf_err_message)
 462              CALL wrf_debug ( 0, '---- ERROR: aero_opt does not match, comparing namelist to the input file' )
 463              count_fatal_error = count_fatal_error + 1
 464           END IF 
 465        END IF
```

Some code would need to get into output_wrf.F to actually write out the metadata. This is just a code snippet.